src/containers/single-event/SingleEvent.tsx

```diff
@@ -0,0 +1,77 @@
+import { Spin } from 'antd';
+import React, { useEffect } from 'react';
+import { Helmet } from 'react-helmet';
+import { connect, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import styled from 'styled-components';
+import EventDetails from '../../components/event-details/EventDetails';
+import { C4CState } from '../../store';
+import { AsyncRequest, AsyncRequestKinds } from '../../utils/asyncRequest';
+import { getUpcomingEvents } from '../upcoming-events/ducks/thunks';
+import { EventProps, EventsReducerState } from '../upcoming-events/ducks/types';
+
+const ContentContainer = styled.div`
+  padding: 24px;
+  margin: auto;
+  max-width: 1200px;
+  width: 80%;
+`;
+
+const CenteredContainer = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+interface UpcomingEventsProps {
+  readonly events: EventsReducerState['upcomingEvents'];
+}
+
+const SingleEvent: React.FC<UpcomingEventsProps> = ({ events }) => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (events.kind === AsyncRequestKinds.NotStarted) {
+      dispatch(getUpcomingEvents());
+    }
+  }, []);
+
+  const id: number = Number(useParams().id);
+
+  const conditionalRenderEventDetails = (
+    eventsList: AsyncRequest<EventProps[], any>,
+  ) => {
+    if (eventsList.kind === AsyncRequestKinds.Completed) {
+      const event = eventsList.result.filter((e) => e.id === id);
+
+      if (event.length > 0) {
```

Comment:
Could we perhaps use reduce instead of filter to return a single item? If not, I think it makes more sense to have event.length === 1 and throw an error in other cases.

Reply:
I font think using reduce instead of filter makes a difference here, I don't think we need to worry about the > 1 event case, and even then we probably wouldn't want to error out, but display any of the events.

Reply:
I more meant just syntactically you could reduce to an `EventProps?` as opposed to `EventProps[]`, but not a huge difference either way.